### PR TITLE
Fixed compiler version 10. support with minimal changes

### PR DESCRIPTION
**Pull request by PieterValentijn**
Fixed compiler version 10. support with minimal changes and added installlibs.bat for easy download of dependency.

### CODE REVIEW

File: installlibs.bat
Comment (svanas) [inline installlibs.bat:1]: nothing wrong with this file if the user happens to execute it from within the directory where Delphereum got cloned into, but I'd rather have them follow [these instructions](https://svanas.medium.com/connecting-delphi-to-a-local-in-memory-blockchain-9a1512d6c5b0) and clone manually.

File: web3.coincap.pas [hunk 1]
```diff
@@ -86,7 +86,12 @@   TAssets = class(TDeserializedArray<IAsset>)
 
 function TAssets.Item(const Index: Integer): IAsset;
 begin
+{$IF CompilerVersion < 40.0}
+  Result := TAsset.Create(TJsonArray(FJsonValue).Items[Index]);
+{$IFEND}
+{$IF CompilerVersion >= 40.0}
```
Comment (svanas) [inline web3.coincap.pas:92]: is there no `{$IFELSE}` we can use here?

File: web3.json.pas [hunk 2]
```diff
@@ -210,9 +210,14 @@ function getPropAsUInt64(const obj: TJsonValue; const name: string; const def: U
   const P = TJsonObject(obj).Get(name);
   if Assigned(P) and Assigned(P.JsonValue) then
     if P.JsonValue is TJsonNumber then
-      Result := TJsonNumber(P.JsonValue).AsUInt64
+{$IF CompilerVersion < 40.0}
+      Result := TJsonNumber(P.JsonValue).AsInt64
+{$IFEND}
+{$IF CompilerVersion >= 40.0}
```
Comment (svanas) [inline web3.json.pas:216]: See earlier comment about `{$IFELSE}`

File: web3.json.pas [hunk 3]
```diff
@@ -210,9 +210,14 @@ function getPropAsUInt64(const obj: TJsonValue; const name: string; const def: U
...
+        Result := TJsonNumber(P.JsonValue).AsUInt64
+{$IFEND}
     else if P.JsonValue is TJsonString then
-      Result := StrToUInt64Def(TJsonString(P.JsonValue).Value, def);
+      Result := StrToUInt64Def(TJsonString(P.JsonValue).value, def);
```
Comment (svanas) [inline web3.json.pas:220]: Why did the case of this property got changed?

Comment (svanas) [reply web3.json.pas]: I'm unsure why `Value` (capitalized) got changed into `value` (lowercase).

File: web3.coincap.pas [hunk 4]
```diff
@@ -86,7 +86,11 @@   TAssets = class(TDeserializedArray<IAsset>)
 
 function TAssets.Item(const Index: Integer): IAsset;
 begin
+  {$IF CompilerVersion < 35}
+  Result := TAsset.Create(TJsonArray(FJsonValue).Items[Index]);
+  {$ELSEIF CompilerVersion >= 35}
```
Comment (svanas) [inline web3.coincap.pas:91]: Wouldn't an `{$ELSE}` be sufficient here?

File: web3.json.pas [hunk 5]
```diff
@@ -210,7 +210,11 @@ function getPropAsUInt64(const obj: TJsonValue; const name: string; const def: U
   const P = TJsonObject(obj).Get(name);
   if Assigned(P) and Assigned(P.JsonValue) then
     if P.JsonValue is TJsonNumber then
+      {$IF CompilerVersion < 35}
+      Result := StrToUInt64Def(P.JsonValue.Value, def)
+      {$ELSEIF CompilerVersion >= 35}
```
Comment (svanas) [inline web3.json.pas:215]: See comment about `{$ELSE}` above.

File: web3.json.pas (hunk 3 above)
Comment (PieterValentijn) [reply web3.json.pas]: it was me using Ctrl + D.
I have removed that.

File: web3.json.pas (hunk 5 above)
Comment (PieterValentijn) [reply web3.json.pas]: Ok done.

**Pull request merged**